Website/src/rendering/src/components/PreviewSearch/PreviewSearch.tsx

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import ClickOutside from './ClickOutside';
+import LeftColumn from './LeftColumn';
+import RightColumn from './RightColumn';
+
+function debounce(
+  func: (arg: string | { value: string }) => void,
+  wait: number,
+  immediate: boolean
+) {
+  let timeout: NodeJS.Timeout;
+
+  return function returnFn(this: unknown, ...rest: unknown[]) {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const context = this;
+
+    const args = rest;
+
+    const later = function executeFn() {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
+    };
+
+    const callNow = immediate && !timeout;
+
+    clearTimeout(timeout);
+    timeout = setTimeout(later, wait);
+    if (callNow) func.apply(context, args);
```

Comment:
Please always add curly braces after if/else statements even if there is just one instruction to execute.